File: 3059 − Speed Limit/Main.java
import java.util.Arrays;

public class Main {
    Main(){
        int[][][] input = {
            { {3}, {20, 2}, {30, 6}, {10, 7} },
            { {2}, {60, 1}, {30, 5} },
            { {4}, {15, 1}, {25, 2}, {30, 3}, {10, 5} }
        };

        Arrays.asList(input)
              .forEach(i -> 
                calculate(i[0][0], Arrays.copyOfRange(i, 1, i[0][0] + 1))
              );
    }

    /**
     * n = number of pairs of values
     * st = speed in miles per hour, total elapsed time in hours
     */
    void calculate(int n, int[][] st){
        int speedInMiles = st[0][0];
        int elapsedTime = st[0][1];

        int totalMiles = speedInMiles * elapsedTime;

        for (int i = 1; i < n; i++){
            speedInMiles = st[i][0];
            elapsedTime = st[i][1] - st[i - 1][1];
            totalMiles += speedInMiles * elapsedTime;
        }

        System.out.println(totalMiles + " miles");
    }


    public static void main(String[] args) { new Main(); }
}

File: 3059 − Speed Limit/main.py
def milesDriven():
  x = 0
  while (x == 0): # To allow continous inputs
    pairs = int(input("How many pairs of data? "))
    if (pairs == -1): # When -1 is input, to end the function
      return
    mphTotal = []
    timeTotal = []
    distance = 0
    elapsedTime = 0
    
    for i in range(0, pairs):
      mph = int(input("What was the mph for the pairs (enter individually) "))
      mphTotal.append(mph)
      time = int(input("Enter time recorded at that speed (in hours) "))
      timeTotal.append(time)
    for i in range(0, pairs):
      if (timeTotal[i-1] > timeTotal[i]): # To fix an error where i-1 would backflow into last number in array
        elapsedTime = timeTotal[i] 
      else:
        elapsedTime = timeTotal[i] - timeTotal[i-1]
      distance = distance + (mphTotal[i] * elapsedTime) # Calculations for each distance / time pair
    print(distance, "miles")

milesDriven()
